time/ptp.md

```diff
@@ -0,0 +1,218 @@
+
+#### PTP Clock Types
+
+Clocks synchronized by PTP are organized under a hierachy of master-slave type.
+The slaves are synchronized to their masters who can themselves be slaves to
+other masters. When a clock has only one port, it can be master or slave, this
+type of clock is called an `Ordinary Clock`, A clock with multiple ports can be
+master on one port and slave on another, this type of clock is called a
+`Boundary Clock`. The PTP Clock type supported in this proposal is
+`Ordinary Clock` and only one network interface on each node can be configured
+for PTP.
+
+#### Node PTP Config
+
+Although multiple `NodePTPCfg` custom resources can be defined in the cluster,
+only one profile can be applied to a specific node. Each profile contains
+combination of `Interface`,  `ptp4lOpts` and `phc2sysOpts`, one clock at most
+can be configured on each node, this excludes the possibility to configure an
+OpenShift node to be a PTP `Boundary Clock` node.
+
+#### PTP Network Transport Mode
+
+PTP protocol supports three transport modes: IEEE 802.3, UDP IPv4 and UDP IPv6.
+When using UDP network transport mode, PTP device requires IP address be
+configured, however this is not supported by PTP daemon in this proposal, user
+will need to configure it manually. `NodePTPCfg` might be extended in the future
+to support necessary network configurations such as creating a bond interface
+for PTP redundancy, assigning IP address for UDP transport mode.
+
+#### PTP4L configuration file
+
+Default ptp4l configuration file (ptp4l.conf) will be used when starting ptp4l
+service, user doesn't need to specify ptp4l.conf in `ptp4lOpts`. The content
+of ptp4l.conf is using default file delivered in linuxptp package. This
+configuration file cannot be changed via CRDs exposed in this proposal. However,
+some of ptp4l configuration options in ptp4l.conf can be overwrited by
+specifying `ptp4lOpts` in `NodePTPCfg` custom resource.
+
+#### Redundancy
+
+##### PTP port redundancy
+
+Ptp4l supports using bonding interface in active-backup mode. It uses the
+active interface from bond as PTP clock and can switch to another active
+interface in case of a failover. Creation of bonding interface(e.g. bond0)
+is not supported in this proposal, to use bond interface as PTP device,
+user needs to create it manually(for example, on RHEL nodes).
+
+##### PTP and NTP redundancy
+
+Chrony is deployed and enabled by default during OpenShift installation process.
+This proposal doesn't support timemaster service configuration. Use of PTP will
+require disabling NTP(Chrony) first. No redundancy is provided between NTP and
+PTP in this proposal.
+
+### Risks and Mitigations
+
+In case of failure in linuxptp configuration, OpenShift nodes will be left as
+no time source to sync, resulting in potential time disorder among nodes.
+This can be resolved by providing PTP and NTP redundancy via timemaster service
+which automatically rolls back to use default NTP time source if configured.
```

Comment:
This looks like we could mitigate the risk of failure by using timemaster (which is right) but we don't support timemaster service configuration. Just to make it clear.

Reply:
updated to explicitly mention :
1) timemaster service can mitigate the redundancy issue between PTP and NTP
2) timemaster configuration will not be supported in initial proposal.

Reply:
LGTM